submissions/src/test/kotlin/SubmissionControllerTest.kt

```diff
@@ -0,0 +1,290 @@
+package gov.cdc.prime.reportstream.submissions.controllers
+
+import com.azure.data.tables.TableClient
+import com.azure.storage.blob.BlobClient
+import com.azure.storage.blob.BlobContainerClient
+import com.azure.storage.blob.BlobServiceClient
+import com.azure.storage.queue.QueueClient
+import com.azure.storage.queue.QueueServiceClient
+import com.azure.storage.queue.models.SendMessageResult
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.microsoft.applicationinsights.TelemetryClient
+import gov.cdc.prime.reportstream.submissions.config.AzureConfig
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.ArgumentMatchers.any
+import org.mockito.ArgumentMatchers.anyLong
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.Mockito.doNothing
+import org.mockito.Mockito.doThrow
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.`when`
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers
+import java.io.ByteArrayInputStream
+import org.junit.jupiter.api.AfterEach
+import org.mockito.ArgumentMatchers.anyMap
+import org.mockito.ArgumentMatchers.isNull
+import org.mockito.Mockito.reset
+import org.mockito.MockitoAnnotations
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.whenever
+import org.springframework.context.annotation.Import
+
+@WebMvcTest(SubmissionController::class)
+@Import(AzureConfig::class)
+class SubmissionControllerTest {
+
+    @Autowired
+    private lateinit var mockMvc: MockMvc
+
+    @MockBean
+    private lateinit var blobServiceClient: BlobServiceClient
+
+    @MockBean
+    private lateinit var queueServiceClient: QueueServiceClient
+
+    @MockBean
+    private lateinit var tableClient: TableClient
+
+    @MockBean
+    private lateinit var telemetryClient: TelemetryClient
+
+    private lateinit var objectMapper: ObjectMapper
+
+    private lateinit var blobContainerClient: BlobContainerClient
+    private lateinit var blobClient: BlobClient
+    private lateinit var queueClient: QueueClient
+    private lateinit var sendMessageResult: SendMessageResult
+
+    @BeforeEach
+    fun setUp() {
+        MockitoAnnotations.openMocks(this)
```

Comment:
Any reason to use mockito over mockk?  Might just be confusing going between the projects

Reply:
I can't remember the specifics now. But I started using mockk and hit some issues then all the guides and online advice for sping were using mockito so it felt more inline with Spring. What do you think?